### DISCUSSION TITLE
Great piece of software!

### DISCUSSION THREAD

**Discussion by narrsam** [ensemblesaw/ensembles-app#63] [Q&A]
This is a long awaited open source project by me and I believe a lot of ppl.

I have a bunch of questions:

1. Is this intended to be multi platform? Or just on Linux? If it is multi-platform capable, please include some instructions on how to build it on other OSs.
2. Is it planned to be able to choose a different sound engine than FluidSynth? VSTs? or maybe just use it as an arranger controller for external midi devices / workstations?
3. I would like to have it work on a RaspberryPi with a touch screen, I don't care much for the sound engine, just the midi out for the RraspberryPi to an external device would be great! Is that possible?

Thank you for the great work and keep it up please, I will try to catch-up on Vala and see if I can help.

**Comment by SubhadeepJasu** [comment]
Hi Narrsam!

To answer your questions one by one:
1. It's not planned for any other platform at the moment. Once we have a full release on Linux, I'll explore those options.
2. You wont be able to replace FluidSynth, however there is support for VST, LV2 and native plugins planned. Sadly I am lacking the technical knowhow so I have a half baked implementation of LV2 support in the app right now. Any help in that field is greatly appreciated.
3. You will be able to build this app on Raspberry Pi, and use Jack or PipeWire as the audio driver (for best performance).

A lot of the internal or rather "core" component code is also written in C, if you want to check it out.

**Reply by narrsam** [reply]
Thanks for the swift reply, I am a C# guy so Vala seems like an easier rout :p
What is the best way to setup a project/debugging environment (code editor, debugger, etc...) to be able to make a branch and try working with the code? I'm sorry I'm not familiar with meson and flatpak( I'm used to the MS Visual Studio project setup where you just press the play button :D), is it the only way to run the app from source code? or is there another way to debug it (I tried doing valac main.vala but it is not picking up the namespaces and the other packages the flatpak is providing when using flatpak-builder)

**Reply by SubhadeepJasu** [reply]
I would suggest going through the README. All the instructions are there. For starters, you will need to install meson, flatpak and flatpak builder (if you do flatpak build). If you are on elementary OS, I guess those things should already be available. Apart from that you will need elementary-sdk. Then when you run the meson build command it will prompt you for things that are missing and you will need to install those. Let me know if you get stuck. Installing things like fluidsynth and lv2 can be a tricky. 

If you do flatpak build, its just the two commands in the README.

**Reply by SubhadeepJasu** [reply]
Apart from this I would also suggest installing vala language server and c and c++ tools for vs code for a more Visual Studio like intellisense.
